main.c

```diff
@@ -643,7 +644,25 @@ static void *tcmur_cmdproc_thread(void *arg)
 			 */
 			if (ret != TCMU_STS_ASYNC_HANDLED) {
 				completed = 1;
-				tcmur_command_complete(dev, cmd, ret);
+				tcmur_command_complete(dev, cmd, ret, false);
+			}
+
+			/* setup timer for async handled commands */
+			if (!rhandler->has_timer && ret == TCMU_STS_ASYNC_HANDLED) {
+				struct tcmu_timer *timer;
+
+				timer = calloc(1, sizeof(*timer));
+				if (!timer) {
+					tcmu_dev_err(dev, "malloc timer failed!\n");
+					return NULL;
+				}
+
+				timer->data = cmd;
+				timer->expires = tcmu_get_dev_cmd_time_out(dev) + 5;
+				timer->function = tcmur_cmd_timeout;
+				cmd->timer = timer;
+
+				tcmu_add_timer(timer);
```

Comment:
There will be one problem here too:

If the glfs or other backend quick enough, and just before here the cmd->done() is called then the cmd maybe released, so the cmd deferences will be segment crash.